applications/luci-app-keepalived/htdocs/luci-static/resources/view/keepalived/vrrp_instance.js

```diff
@@ -0,0 +1,286 @@
+
+		o = s.taboption('general', form.Value, 'priority', _('Priority'),
+			_('A server with a higher priority becomes a MASTER'));
+		o.datatype = 'uinteger';
+		o.optional = false;
+		o.rmempty = false;
+
+		o = s.taboption('general', form.Value, 'advert_int', _('Interval'),
+			_('VRRP Advert interval in seconds'));
+		o.datatype = 'float';
+		o.default = '1';
+		o.rmempty = false;
+		o.optional = false;
+		o.placeholder = '1';
+
+		o = s.taboption('general', form.Flag, 'nopreempt', _('Disable Preempt'),
+			_('Allows the lower priority machine to maintain the master role,') +
+			_('even when a higher priority machine comes back online.') + ' ' +
+			_('For this to work, the initial state of this entry must be BACKUP.'));
+		o.default = false;
+		o.rmempty = false;
+
+		ipaddress = uci.sections('keepalived', 'ipaddress');
+		o = s.taboption('general', form.DynamicList, 'virtual_ipaddress', _('Virtual IP Address'),
+			_('Addresses add|del on change to MASTER, to BACKUP.') + ' ' +
+			_('With the same entries on other machines, the opposite transition will be occurring.'));
+		if (ipaddress != '') {
+			for (var i = 0; i < ipaddress.length; i++) {
+				o.value(ipaddress[i]['name']);
+			}
+		}
+		o.rmempty = false;
+		o.optional = false;
+	},
+
+	renderPeerTab: function(s, netDevs) {
+		var o;
+
+		o = s.taboption('peer', form.Value, 'unicast_src_ip', _('Unicast Source IP'),
+			_('Default IP for binding vrrpd is the primary IP on interface'));
+		o.datatype = 'ipaddr';
+		o.optional = true;
+		o.modalonly = true;
+		for (var i = 0; i < netDevs.length; i++) {
+			var addrs = netDevs[i].getIPAddrs();
+			for (var j = 0; j < addrs.length; j++) {
+				o.value(addrs[j].split('/')[0]);
+			}
+		}
+
+		var peers = uci.sections('keepalived', 'peer');
+		o = s.taboption('peer', form.DynamicList, 'unicast_peer', _('Peer'),
+			_('Do not send VRRP adverts over VRRP multicast group.') + ' ' +
+			_('Instead it sends adverts to the following list of ip addresses using unicast design fashion'));
+		if (peers != '') {
+			for (var i = 0; i < peers.length; i++) {
+				o.value(peers[i]['name']);
+			}
+		}
+
+		o = s.taboption('peer', form.Value, 'mcast_src_ip', _('Multicast Source IP'),
+			_('If you want to hide location of vrrpd, use this IP for multicast vrrp packets'));
+		o.datatype = 'ipaddr';
+		o.optional = true;
+		o.modalonly = true;
+		o.depends({ 'unicast_peer' : '' });
+
+		o = s.taboption('peer', form.ListValue, 'auth_type', _('HA Authentication Type'));
+		o.value('PASS', _('Simple Password'));
+		o.value('AH', _('IPSec'));
+
+		o = s.taboption('peer', form.Value, 'auth_pass', _('Password'),
+			_('Password for accessing vrrpd, should be the same on all machines'));
+		o.datatype = 'maxlength(8)';
+		o.password = true;
+		o.modalonly = true;
+		o.depends({ 'auth_type' : 'PASS' });
+	},
+
+	renderGARPTab: function(s) {
+		var o;
+
+		o = s.taboption('garp', form.Value, 'garp_master_delay', _('GARP Delay'),
+			_('Gratuitous Master Delay'));
```

Comment:
are that seconds? If so see maybe  add some common seconds as values, so we have a dropdown

The same comment applies to

- garp_master_repeat
- garp_master_refresh
- garp_master_refresh_repeat

Reply:
Done